app/components/jobseekers/search_results/heading_component.rb

```diff
@@ -4,25 +4,26 @@ def initialize(vacancies_search:)
     @keyword = @vacancies_search.keyword
     @location = @vacancies_search.location_search.location
     @polygon_boundaries = @vacancies_search.location_search.polygon_boundaries
+    @radius = @vacancies_search.search_criteria[:buffer_radius] || @vacancies_search.search_criteria[:radius]
     @total_count = @vacancies_search.total_count
   end
 
   def heading
-    if @keyword.present? && @polygon_boundaries.present? && @vacancies_search.location_search.buffer_radius.nil?
+    if @keyword.present? && @polygon_boundaries.present?
       I18n.t("jobs.search_result_heading.keyword_location_polygon_html",
-             jobs_count: number_with_delimiter(@total_count), location: @location, keyword: @keyword, count: @total_count)
+             jobs_count: number_with_delimiter(@total_count), location: @location, keyword: @keyword, count: @total_count, radius: "#{@radius} #{I18n.t('jobs.search_result_heading.unit_of_length').pluralize(@radius.to_i)}")
```

Comment:
This way looks cleaner - combining the radius and the units in the translation instead of here in the component:

```ruby
jobs_count: number_with_delimiter(@total_count), location: @location, keyword: @keyword, count: @total_count, radius: @radius, units: I18n.t('jobs.search_result_heading.unit_of_length').pluralize(@radius.to_i))
```